src/main/java/net/ccbluex/liquidbounce/injection/mixins/minecraft/entity/MixinPlayerEntity.java

```diff
@@ -209,15 +209,27 @@ private Vec3d hookSlowVelocity(Vec3d instance, double x, double y, double z) {
         return instance.multiply(x, y, z);
     }
 
+    @SuppressWarnings("UnreachableCode")
     @WrapWithCondition(method = "attack", at = @At(value = "INVOKE", target = "Lnet/minecraft/entity/player/PlayerEntity;setSprinting(Z)V", ordinal = 0))
     private boolean hookSlowVelocity(PlayerEntity instance, boolean b) {
         if ((Object) this == MinecraftClient.getInstance().player) {
+            ModuleKeepSprint.INSTANCE.setSprinting(b);
             return !ModuleKeepSprint.INSTANCE.getEnabled() || b;
         }
 
         return true;
     }
 
+    @SuppressWarnings({"UnreachableCode", "ConstantValue"})
+    @ModifyExpressionValue(method = "attack", at = @At(value = "INVOKE", target = "Lnet/minecraft/entity/player/PlayerEntity;isSprinting()Z"))
+    private boolean hookSlowVelocity(boolean original) {
+        if ((Object) this == MinecraftClient.getInstance().player && ModuleKeepSprint.INSTANCE.handleEvents()) {
+            return ModuleKeepSprint.INSTANCE.getSprinting();
```

Comment:
why would you need this?  So you basically get current "isSprinting" status (line 216), keep it even more current by updating it at PlayerPostTickEvent event (??????) and then give "attack" method current is sprinting status. But we didn't change/spoof it. KeepSprinting modifies vector multiplications result, without affecting this variable

Reply:
to prevent flagging anticheats like grim with >20 cps and 0.6 MotionWhenHurt

Reply:
> to prevent flagging anticheats like grim with >20 cps and 0.6 MotionWhenHurt

how would this help? as far as I understand we change isSprinting to it's original value (comment above), but since we didn't modify it we basically don't change anything

Reply:
it not affecting the sprinting variable is the problem, you can't slow yourself twice in one tick